crates/runtime/src/dataconnector/flightsql.rs

```diff
@@ -0,0 +1,127 @@
+use std::collections::HashMap;
+use std::pin::Pin;
+use std::{future::Future, sync::Arc};
+
+use arrow_flight::sql::client::FlightSqlServiceClient;
+use arrow_flight::Ticket;
+use async_trait::async_trait;
+use spicepod::component::dataset::Dataset;
+use tonic::transport::Channel;
+
+use flight_client::tls::new_tls_flight_channel;
+use flightsql_datafusion::FlightSQLTable;
+use secrets::Secret;
+
+use super::DataConnector;
+use arrow::error::ArrowError;
+use futures::stream::TryStreamExt;
+
+#[derive(Debug, Clone)]
+pub struct FlightSQL {
+    pub client: FlightSqlServiceClient<Channel>,
+}
+
+impl FlightSQL {
+    async fn query(
+        client: FlightSqlServiceClient<Channel>,
+        query: String,
+    ) -> Result<Vec<arrow::record_batch::RecordBatch>, Box<dyn std::error::Error>> {
+        let flight_info = client.clone().execute(query, None).await?;
+
+        let mut batches = vec![];
+        for ep in &flight_info.endpoint {
+            if let Some(tkt) = &ep.ticket {
+                match batch_from_ticket(&mut client.clone(), tkt.to_owned()).await {
```

Comment:
This also needs to handle the case where the URL in endpoint isn't None. If so, we need to connect to that new URL to run the do_get.

Reply:
Also we should unify this logic so we're not repeating it - you have a different implementation of `query` in `crates/flightsql_datafusion/src/lib.rs`.